connector/MongoDBAtlasODBC.pq

```diff
@@ -51,13 +52,71 @@ GetCredentialConnectionString = () as record =>
     in
         CredentialConnectionString;
 
-// SQL-1117: Implement ProcessUri Function
-ProcessUri = (uri as text) as record => 
+NullFieldCheck = (field as any, fieldName as text ) =>
+    let 
+        Result = if field <> null then Record.AddField([], fieldName, field) else []
+    in
+        Result;
+
+ProcessAuthSrc = (queryEntry as any, pathEntry as any) =>

```

Comment:
Should we actually bother doing any validation here? We could leave it for the driver to handle. WDYT?